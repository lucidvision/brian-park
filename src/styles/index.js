export const colors = {
  primary: 'rgba(0, 13, 85, 0.8)',
  black: '#000',
  white: '#fff',
  gradientLight: 'rgba(0, 38, 255, 0.8)',
  gradientDark: 'rgba(0, 0, 0, 0.8)'
}

export const fontSizes = {
  large: '6rem',
  medium: '4rem',
  small: '2.2rem',
  default: '1.6rem'
}

export const screenSizes = {
  desktop: '112.5em',
  tablet: '75em',
  phone: '56.25em'
}

export const utilities = {
  marginSmall: '1.5rem',
  marginMedium: '4rem',
  marginLarge: '8rem'
}
