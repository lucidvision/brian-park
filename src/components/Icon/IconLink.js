import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { colors } from 'styles'

IconLink.propTypes = {
  className: PropTypes.string.isRequired,
  color: PropTypes.string,
  url: PropTypes.string.isRequired
}

IconLink.defaultProps = {
  className: '',
  color: colors.primary,
  url: ''
}

const StyledIcon = styled.i`
  font-size: 3em;
  color: ${props => props.color}
`

export default function IconLink (props) {
  return (
    <a href={props.url}>
      <StyledIcon className={props.className} {...props} />
    </a>
  )
}
