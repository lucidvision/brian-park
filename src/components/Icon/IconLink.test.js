import React from 'react'
import { mount } from 'enzyme'
import IconLink from './IconLink'
import { colors } from 'styles'

const createIconLink = (override) => {
  const props = {
    url: 'www.rainbows.com',
    ...override
  }

  return mount(<IconLink {...props} />)
}

describe('IconLink', () => {
  const component = createIconLink()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  it('renders with the correct props', () => {
    expect(component.find('IconLink').props()).toEqual({
      color: colors.primary,
      url: 'www.rainbows.com'
    })
  })

  it('renders a link with the href', () => {
    expect(component.find('a').props()).toMatchObject({
      href: 'www.rainbows.com'
    })
  })

  it('renders an icon with the correct props', () => {
    expect(component.find('i').props()).toMatchObject({
      color: colors.primary
    })
  })
})
