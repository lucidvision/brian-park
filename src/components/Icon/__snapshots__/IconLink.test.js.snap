// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLink renders correctly 1`] = `
<IconLink
  color="rgba(0, 13, 85, 0.8)"
  url="www.rainbows.com"
>
  <a
    href="www.rainbows.com"
  >
    <styled.i
      color="rgba(0, 13, 85, 0.8)"
    >
      <i
        className="sc-bdVaJa cIrqXy"
        color="rgba(0, 13, 85, 0.8)"
      />
    </styled.i>
  </a>
</IconLink>
`;
